shcomplete/tfdf.py

```diff
@@ -0,0 +1,104 @@
+import os
+import re
+import logging
+from collections import Counter
+
+
+def get_all_prefixes(args):
+    """
+    Return a list of list of all the prefixes encountered in the files among the input directory.
+    """
+    all_prefixes = []
+    for root, d, files in os.walk(args.directory):
+        for file_name in files:
+            all_prefixes.append([])
+            path_to_file = os.path.join(root, file_name)
+            with open(path_to_file, "r") as f:
+                for line in f:
+                    line = line.rstrip()
+                    line = line.split(" ")
+                    for j in range(len(line)):
+                        pfx = " ".join(x for x in line[:j+1])
+                        all_prefixes[-1].append(pfx)
+    return all_prefixes
+
+
+def df(args, all_prefixes):
+    """
+    Compute the dociment frequency values of all prefixes in the files among the input directory.
+    """
+    list_prefixes = []
+    for prefixes in all_prefixes:
+        prefixes = list(set(prefixes))
+        list_prefixes += prefixes
+    df_values = Counter(list_prefixes)
+
+    return df_values
+
+
+def tf(prefix, file_name):
+    """
+    Compute the term frequency of prefix in file_name.
+    """
+    with open(file_name, "r") as f:
+        return sum(1 for line in f if prefix in line)
```

Comment:
line.startswith(prefix)

And you need to do the same line processing you do in `get_all_prefixes()`